### ISSUE TITLE
Tests Failing on MinGW (Windows 8.1)

### ISSUE THREAD

**Issue by Choonster** [libuv/libuv#264]
libuv: v1.4.2
OS: MINGW32_NT-6.2 (MSYS shell on Windows 8.1)

Failed tests:
```
`pipe_getsockname_blocking` failed: exit code 3
`spawn_reads_child_path` failed: exit code 3
`fs_utime` failed: exit code 3
`fs_futime` failed: exit code 3
`fs_symlink_dir` failed: exit code 3
`threadpool_multiple_event_loops` failed: timeout
```

[Full test output](https://gist.github.com/Choonster/b42ca82ea7731595ed9c)

Steps to reproduce:
```
git clone https://github.com/libuv/libuv.git
cd libuv
git checkout v1.4.2
sh autogen.sh
./configure
make
make check &>testoutput.txt
```

Let me know if you need any more information.

**Comment by txdv** [comment]
I'll look into, pipe_getsockname_blocking, it is probably related to what https://github.com/libuv/libuv/issues/246

**Comment by martell** [comment]
I tested all the release versions and have determined
pipe_getsockname_blocking was broken between 1.2.1 and 1.3.0

**Comment by martell** [comment]
@saghul @txdv 

I am one of the maintainers of msys2.
It essentially is a linux distro for windows
We use the arch linux package manager pacman

So I created a patch to revert that commit and it fixes the testcase but I think we need a proper fix
https://github.com/Alexpux/MINGW-packages/tree/master/mingw-w64-libuv

Could you investigate ?

**Comment by saghul** [comment]
I'm a msys2 user myself :-) I'll try to find some time to investigate.

**Comment by vtjnash** [comment]
I looked into this and think that it is a bug in the test code. Due to https://github.com/libuv/libuv/pull/166/files#diff-7be27b71e7f7a6d2c40c447747db1231R2035, it is expected that `len == 0`, so this new test has been doing an OOB memory read of `buf[len - 1]`

**Comment by saghul** [comment]
Can you send a PR?

**Comment by vtjnash** [comment]
Yeah, I need to finally move a number of local fixes to pull requests (rebasing https://github.com/JuliaLang/libuv/commits/julia-uv1.9.0 is a bit cumbersome).

For cross-reference, the `pipe_getsockname_blocking` bug is also https://github.com/libuv/libuv/issues/376 and https://github.com/libuv/libuv/issues/529

**Comment by saghul** [comment]
Cool, we can kill a few birds with a single stone then :-)

**Comment by saghul** [comment]
I landed a PR which fixes the pipe_getsockname_blocking failure: #981 . Since this has been around for quite a while, I'm going to close it. Please open a new issue if tests are failing with latest v1.x branch.
